common/thorhelper/thorsoapcall.cpp

```diff
@@ -1040,32 +1092,35 @@ class CWSCHelper : implements IWSCHelper, public CInterface
             }
             StringBuffer secretName("http-connect-");
             secretName.append(finger);
-            Owned<IPropertyTree> secret = (vaultId.isEmpty()) ? getSecret("ecl", secretName) : getVaultSecret("ecl", vaultId, secretName, nullptr);
-            if (!secret)
-                throw MakeStringException(0, "%sCALL %s SECRET not found", wscType == STsoap ? "SOAP" : "HTTP", secretName.str());
-
-            StringBuffer url;
-            getSecretKeyValue(url, secret, "url");
-            if (url.isEmpty())
-                throw MakeStringException(0, "%sCALL %s HTTP SECRET must contain url", wscType == STsoap ? "SOAP" : "HTTP", secretName.str());
-            UrlListParser urlListParser(url);
-            StringBuffer auth;
-            getSecretKeyValue(auth, secret, "username");
-            if (auth.length())
-            {
-                if (strchr(auth, ':'))
-                    throw MakeStringException(0, "%sCALL HTTP-CONNECT SECRET username contains illegal colon", wscType == STsoap ? "SOAP" : "HTTP");
-                auth.append(':');
-                getSecretKeyValue(auth, secret, "password");
-            }
-            urlListParser.getUrls(urlArray, auth);
-            proxyAddress.set(secret->queryProp("proxy"));
-            getSecretKeyValue(proxyAddress.clear(), secret, "proxy");
+
+            loadConnectSecret(vaultId, secretName, urlArray, issuer, proxyAddress, true, wscType);
         }
         else
         {
             UrlListParser urlListParser(hosts);
             urlListParser.getUrls(urlArray);
+            if (mapUrlsToSecrets && urlArray.length())
+            {
+                StringBuffer secretName;
+                UrlArray tempArray;
+                //TBD: If this is a list of URLs do we A. not check for a mapped secret, B. check the first one, C. Use long secret name including entire list
+                Url &url = urlArray.tos();
+                url.getDynamicUrlSecretName(secretName);
+                if (secretName.length())
+                {
+                    if (loadConnectSecret(nullptr, secretName, tempArray, issuer, proxyAddress, false, wscType))
```

Comment:
Add an option to require all soapcalls use secrets (and pass true)?

Reply:
Done.